components/script/dom/htmltemplateelement.rs

```diff
@@ -58,9 +65,37 @@ impl HTMLTemplateElement {
         n.upcast::<Node>().set_weird_parser_insertion_mode();
         n
     }
+
+    pub(crate) fn set_contents(&self, document_fragment: Option<&DocumentFragment>) {
+        self.contents.set(document_fragment);
+    }
 }
 
 impl HTMLTemplateElementMethods<crate::DomTypeHolder> for HTMLTemplateElement {
+    // <https://html.spec.whatwg.org/multipage/#dom-template-shadowrootmode>
+    make_getter!(ShadowRootMode, "shadowrootmode");
```

Comment:
This should be `make_enumerated_getter!()`. See example for the `dir` attribute:

https://github.com/servo/servo/blob/c844ed232a86fef873dc661de60e282fdb86c0d3/components/script/dom/htmlelement.rs#L163